src/intTest/java/uk/nhs/digital/nhsconnect/nhais/jms/InboundMeshServiceInterchangeTest.java

```diff
@@ -0,0 +1,100 @@
+package uk.nhs.digital.nhsconnect.nhais.jms;
+
+import lombok.SneakyThrows;
+import org.assertj.core.api.SoftAssertions;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.r4.model.Parameters;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.Resource;
+import org.springframework.test.annotation.DirtiesContext;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.ReferenceTransactionType;
+import uk.nhs.digital.nhsconnect.nhais.model.mesh.MeshMessage;
+import uk.nhs.digital.nhsconnect.nhais.model.mesh.WorkflowId;
+import uk.nhs.digital.nhsconnect.nhais.parse.FhirParser;
+import uk.nhs.digital.nhsconnect.nhais.repository.DataType;
+import uk.nhs.digital.nhsconnect.nhais.repository.InboundState;
+import uk.nhs.digital.nhsconnect.nhais.service.InboundMeshService;
+import uk.nhs.digital.nhsconnect.nhais.service.TimestampService;
+import uk.nhs.digital.nhsconnect.nhais.utils.JmsHeaders;
+import uk.nhs.digital.nhsconnect.nhais.utils.OperationId;
+
+import javax.jms.JMSException;
+import javax.jms.Message;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.time.Instant;
+import java.time.ZonedDateTime;
+
+@DirtiesContext
+public class InboundMeshServiceInterchangeTest extends InboundMeshServiceBaseTest {
```

Comment:
I'd change the class name to something like `InboundMeshServiceRegistrationTest` since every EDIFACT file has an interchange